storage/mysql/storage_test.go

```diff
@@ -552,6 +552,82 @@ func TestDequeueLeavesGuardInterval(t *testing.T) {
 	}
 }
 
+func TestDequeueLeavesTimeOrdering(t *testing.T) {
+	// Queue two small batches of leaves at different timestamps. Do two separate dequeue
+	// transactions and make sure the returned leaves are respecting the time ordering of the
+	// queue.
+
+	logID := createLogID("TestDequeueLeavesTimeOrdering")
+	db := prepareTestLogDB(logID, t)
+	defer db.Close()
+	s := prepareTestLogStorage(logID, t)
+	batchSize := 2
+
+	{
+		tx := beginLogTx(s, t)
+		defer failIfTXStillOpen(t, "TestDequeueLeavesTimeOrdering", tx)
```

Comment:
No, they should all get committed by the end of the test. But I missed the defer for tx3 so I added that one.